src/directives/rcdni.js

```diff
@@ -0,0 +1,42 @@
+directives.directive('rcDni',['dniSemanticsValidationService','dniSyntaxValidationService',function(dniSemanticsValidationService,dniSyntaxValidationService){
+	return{
+		require: 'ngModel',					
+		link: function (scope, elm, attrs, ctrl){	
+			var validationToExecute = function (viewValue){
+				var currentTypedValue = viewValue;
+				validateDNI(currentTypedValue);
```

Comment:
You can just send viewValue to validateDNI function and avoid creating a new variable.